src/tests/resource-listers/AllResourcesLister.test.js

```diff
@@ -0,0 +1,210 @@
+            { TableNames: [ 'table-d', 'table-e', 'table-f' ], LastEvaluatedTableName: 'further-table-set' },
+            { TableNames: [ 'table-g', 'table-h', 'table-i' ] },
+         ]));
+
+         return lister._fetchTableNames()
+            .then(function(tableNames) {
+               sinon.assert.calledOnce(stub);
+               sinon.assert.calledOn(stub, lister);
+               sinon.assert.calledWith(stub, dynamo, 'listTables', {}, 'ExclusiveStartTableName', 'LastEvaluatedTableName');
+
+               expect(tableNames).to.eql([
+                  'table-a', 'table-b', 'table-c',
+                  'table-d', 'table-e', 'table-f',
+                  'table-g', 'table-h', 'table-i',
+               ]);
+
+               stub.restore();
+            });
+      });
+
+   });
+
+   describe('_fetchAll', function() {
+
+      it('works when there are only one page of results', function() {
+         var service = { action: function() { } },
+             actionStub = sinon.stub(service, 'action'),
+             expectedResponse = { data: 'value' };
+
+         actionStub.callsArgWithAsync(1, null, expectedResponse);
+
+         return lister._fetchAll(service, 'action', { search: 'param' }, 'requestToken', 'responseToken')
+            .then(function(resps) {
+               sinon.assert.calledOnce(actionStub);
+               sinon.assert.calledOn(actionStub, service);
+               sinon.assert.calledWith(actionStub, { search: 'param', requestToken: null });
+
+               expect(resps).to.be.an('array');
+               expect(resps).to.have.length(1);
+               expect(resps[0]).to.eql(expectedResponse);
+
+               actionStub.restore();
+            });
+      });
+
+      it('works when there is more than one page of results', function() {
+         var service = { action: function() { } },
+             actionStub = sinon.stub(service, 'action'),
+             expectedResponse1 = { data: 'some stuff', responseToken: 'get-more' },
+             expectedResponse2 = { data: 'more stuff' };
+
+         actionStub.onCall(0).callsArgWithAsync(1, null, expectedResponse1);
+         actionStub.onCall(1).callsArgWithAsync(1, null, expectedResponse2);
+
+         return lister._fetchAll(service, 'action', { search: 'param' }, 'requestToken', 'responseToken')
+            .then(function(resps) {
+               sinon.assert.calledTwice(actionStub);
+               sinon.assert.alwaysCalledOn(actionStub, service);
+
+               sinon.assert.calledWith(actionStub, { search: 'param', requestToken: null });
+               sinon.assert.calledWith(actionStub, { search: 'param', requestToken: 'get-more' });
+               sinon.assert.callOrder(
+                  actionStub.withArgs({ search: 'param', requestToken: null }),
+                  actionStub.withArgs({ search: 'param', requestToken: 'get-more' })
+               );
```

Comment:
I was thinking the same thing, but decided to follow the existing assertion format. However, we should be safe to ded-upe these assertions.

According to the sinon docs, [`sinon.assert.callOrder(...)`](http://sinonjs.org/releases/v1.17.6/assertions/#sinonassertcallorderspy1-spy2-) passes if provided spies were called in the specified order. When using [`spy.withArgs(...)`](http://sinonjs.org/releases/v1.17.6/spies/#spywithargsarg1-arg2-), a spy is only considered to be called when "the received arguments match those passed to withArgs". Therefore, if the spy for a function is called with the wrong arguments, it will not pass. This in turn will cause `sinon.assert.callOrder` to fail as all its spies were not called.


**Note:** The doc links are to v1.17.6. This project is locked to using v1.17.5. However, [it doesn't appear](https://github.com/sinonjs/sinon/compare/v1.17.5...v1.17.6) that anything has changed in these assertions between the versions.